diff -r -N -u4 Csound-4.15.src.old/sort.h Csound-4.15.src.new/sort.h
--- Csound-4.15.src.old/sort.h	Fri Aug 18 21:03:04 2000
+++ Csound-4.15.src.new/sort.h	Mon Nov  5 18:55:16 2001
@@ -4,13 +4,8 @@
 #endif
 #define SP ' '
 #define LF '\n'
 
-typedef struct memblk {
-	struct memblk *nxtmem;
-	char   *memend;
-} MEMHDR;
-
 typedef struct srtblk {
 	struct srtblk *nxtblk;
 	struct srtblk *prvblk;
 	short   insno;
